testutils/file_utils.go

```diff
@@ -28,6 +28,30 @@ func BuildTempModule(tb testing.TB, dir string) (string, error) {
 	return modPath, nil
 }
 
+// BuildTempModuleNew will run "go build ." in the provided RDK directory and return the
+// path to the built temporary file. This function will fail if there are any
+// build-related errors.
+//
+// TODO: replace all instances of `BuildTempModule` with this function!
```

Comment:
I'm gonna go ahead and do this as part of this PR, but might wait for an implicit approval for the rest of the change, only because this might be end up being noisy.

I'm basically continuing to propagate [this pattern](https://sourcegraph.com/blog/go/advanced-testing-in-go#test-helpers) as I have for other test helpers.

Reply:
here's [the resulting diff](https://github.com/maximpertsov/rdk/compare/RSDK-6770-lock-parallel-mod...maximpertsov:rdk:RSDK-6770-lock-parallel-mod-test-helper?expand=1)

Reply:
That seems fine to me! I assumed you'll just modify `BuildTempModule` above? Do we still want to mark it `Helper` if we're actively failing inside it (I kind of forget how `Helper` works)?

Reply:
Yes I was planning to commit the linked diff which replaces `BuildTempModule` with my `...New` definition.

We still want `t.Helper()` in this case I think - it makes so the test runner reports cleaner by reporting the line where the helper was called vs the line inside the helper whenever we have a failure. [This article](https://www.alexedwards.net/blog/easy-test-assertions-with-go-generics) had a quote that spells it out a little better than the golang docs:

> The [t.Helper()](https://pkg.go.dev/testing#T.Helper) function indicates to the Go test runner that our Equal() function is a test helper. This means that when t.Errorf() is called from our Equal() function, the Go test runner will report the filename and line number of the code which called our Equal() function in the output.